modules/core/04-channel/types/version_test.go

```diff
@@ -0,0 +1,43 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/ibc-go/modules/core/04-channel/types"
+)
+
+func TestSplitVersions(t *testing.T) {
+	testCases := []struct {
+		name       string
+		version    string
+		mwVersion  string
+		appVersion string
+	}{
+		{
+			"single wrapped middleware",
+			"fee29-1:ics20-1",
+			"fee29-1",
```

Comment:
nit: should we use the actual types.Version here?

Reply:
The merge function won't lead to circular testing since no code imports `types_test`

Using the raw strings sounds fine though